### PR TITLE
Add some Conemu options

### PR DESCRIPTION
**Pull request by pmiossec**
to be able to choose:
* the style of the console (because I prefer the default style than the "Solarize light" ;) )
* the shell to launch (bash, cmd or powershell)
* the size of the font (#3392)

PS: I don't know why but everytime I open a form (with VS2015), I have a lot of changes that are made :(
That's explain the first commit...

### CODE REVIEW

**Comment by jbialobr** [comment]
>I don't know why but everytime I open a form (with VS2015), I have a lot of changes that are made :(

The last person who edited the form had a screen of different dimensions from yours. The VS designer updates the values to match your screen size.

File: GitUI/CommandsDialogs/SettingsDialog/Pages/DetailedSettingsPage.Designer.cs [hunk 1]
```diff
@@ -138,21 +144,23 @@ private void InitializeComponent()
...
             this.tableLayoutPanel2.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.tableLayoutPanel2.RowStyles.Add(new System.Windows.Forms.RowStyle());
-            this.tableLayoutPanel2.Size = new System.Drawing.Size(974, 452);
+            this.tableLayoutPanel2.Size = new System.Drawing.Size(1303, 877);
             this.tableLayoutPanel2.TabIndex = 2;
             // 
             // DetailedSettingsPage
             // 
-            this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
+            this.AutoScaleDimensions = new System.Drawing.SizeF(7F, 16F);
```
Comment (KindDragon) [inline GitUI/CommandsDialogs/SettingsDialog/Pages/DetailedSettingsPage.Designer.cs]: This because you have different system DPI/font size

**Comment by pmiossec** [comment]
In fact, I let the message here (because I fail to revert the noisy changes in this PR) but the **REAL** problem occured when doing my previous PR on the `RevisionGrid` component.
Changes on the file done by the designer break the build :(
In fact, that was the real problem that I couldn't explain....

**Comment by KindDragon** [comment]
BTW we preffer to use `AutoScaleMode = AutoScaleMode.DPI`

**Comment by pmiossec** [comment]
>BTW we preffer to use AutoScaleMode = AutoScaleMode.DPI

Shouldn't we replace the 23 remaining use of `AutoScaleMode.Font`?

**Comment by KindDragon** [comment]
We should change that in designer, replace produce wrong result

**Comment by asherber** [comment]
Clean build of branch now gives me an exception when I click on the Console tab. (Installed 2.49 is fine.)
image

**Comment by asherber** [comment]
Running this branch gives me the following default values:

image

I'm not sure what 'Default' is in the console style, but I think it should say 'Solarized Light' on first run (since I think that's what 2.49 uses). And 'Font size' should be populated with 12

**Comment by asherber** [comment]
As a user, I find the UI of e3b79 to be confusing. Without having read this conversation, I don't think someone would understand what the value in parentheses represent. I think the expectation is that the values in the edit controls should represent what's actually happening. It might be okay for the values in Local/Distributed/Global to be blank if nothing has been entered there, but on a fresh start I think the Effective values should display Solarized Light, bash, and 12.

**Comment by pmiossec** [comment]
>but on a fresh start I think the Effective values should display Solarized Light, bash, and 12.

I missed that. I agree with you it should be better....

**Comment by EbenZhang** [comment]
sorry, linked the wrong pull request due to github's autocompletion....

**Comment by vbjay** [comment]
@EbenZhang I think you can delete the comment that you did to link and the
link might go away.

File: GitCommands/Settings/BasicSettingTypes.cs [hunk 2]
```diff
@@ -43,17 +43,13 @@ public BoolNullableSetting(string aName, SettingsPath settingsSource, bool aDefa
             }
         }
 
-        public bool ValueOrDefault
+        public new bool ValueOrDefault
```
Comment (RussKie) [inline GitCommands/Settings/BasicSettingTypes.cs]: why do you need it at all?

Comment (jbialobr) [inline GitCommands/Settings/BasicSettingTypes.cs]: To change the type from bool? to bool.

**Comment by jbialobr** [comment]
Rebased on master and pushed.
I implemented showing default values on Effective page as suggested in comment https://github.com/gitextensions/gitextensions/pull/3403#issuecomment-267856469.

**Comment by pmiossec** [comment]
Thanks a lot for having done the remaining work required to merge it. I planned to do it in the last few days but didn't found the time :-( I just found enough time to rebase it yesterday ;-)

**Comment by jbialobr** [comment]
Thanks for the PR, the work I did was not related to the PR subject - it was a bug in editing settings in general.

**Pull request closed without merging**